tests/apollo/test_skvbc_state_transfer.py

```diff
@@ -0,0 +1,215 @@
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=initial_primary)
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(without=delayed_replicas),
+            checkpoint_num=1,
+            verify_checkpoint_persistency=False
+        )
+        checkpoint_after = await bft_network.wait_for_checkpoint(replica_id=initial_primary)
+
+        self.assertNotEqual(checkpoint_before, checkpoint_after)
+
+        await self._send_requests_to_all_replicas(5, bft_network, skvbc)
+
+        print("DEBUG: Starting replicas {} in view {}.".format(delayed_replicas, actual_view))
+
+        bft_network.start_replicas(delayed_replicas)
+
+        await self._send_requests_to_all_replicas(5, bft_network, skvbc)
+
+        await bft_network.wait_for_state_transfer_to_start()
+
+        await self._send_requests_to_all_replicas(20, bft_network, skvbc)
+
+        for v in delayed_replicas:
+            await bft_network.wait_for_state_transfer_to_stop(initial_primary,
+                                                              v)
+        for v in delayed_replicas:
+            vc_msgs = await self._get_total_view_change_msgs(v, bft_network)
+            self.assertEqual(vc_msgs, 0)
+        
+    @with_trio
+    @with_bft_network(start_replica_cmd,
+                      selected_configs=lambda n, f, c: f >= 2)
+    async def test_bug_2513_with_adversary(self, bft_network):
+        """
+        The goal of this test is to validate that during state transfer
+        in a Replica no View Change is generated, even while clients
+        send requests to all Replicas.
+
+        1) Start all replicas and insulate 2 of them (this test is valid only for f>=2).
+        2) Initiate Client requests and reach a Checkpoint.
+        3) Release insulated replicas.
+        4) Verify State transfer is initiated.
+        5) Send Client requests to all Replicas.
+        6) Verify no View Change msgs were sent from delayed replicas.
+        """
+        initial_primary = 0
+        insulated_replicas = {1,2}
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        for c in bft_network.clients.values():
+            await c.bind()
+        
+        with net.NodesInsulatingAdversary(bft_network, insulated_replicas) as adversary:
+            adversary.interfere()
+            bft_network.start_all_replicas()
+
+            first_primary = await bft_network.get_current_primary()
+            initial_view = await bft_network.get_current_view()
+            print("DEBUG: primary={} view={}.".format(first_primary, initial_view))
+
+            actual_view = 0
+            timeout = 60
+            with trio.fail_after(timeout):  # seconds
+                async with trio.open_nursery() as nursery:
+                    nursery.start_soon(
+                        self._send_requests_to_all_replicas, timeout, bft_network, skvbc)
+                    #See if replica 1 has become the new primary
```

Comment:
this comment is redundant with the content of the `err_msg`. I'd suggest removing the comment.